core/controllers/email_dashboard.py

```diff
@@ -157,3 +161,40 @@ def get(self):
         }
 
         self.render_json(data)
+
+
+class EmailDashboardResult(base.BaseHandler):
+    """Handler for email dashboard result page."""
+    @require_valid_sender
+    def get(self, query_id):
+        query_model = user_models.UserQueryModel.get(query_id, strict=False)
+        if (query_model is None or
+                query_model.query_status != feconf.USER_QUERY_STATUS_COMPLETED
+                or query_model.submitter_id != self.user_id):
+            raise self.InvalidInputException('400 Invalid query id.')
+
+        self.values.update({
+            'query_id': query_id,
```

Comment:
We are actually trying to get rid of Jinja. I suggest doing the usual thing of defining a corresponding EmailDashboardResultHandler class with get() and post() methods, similar to what we do elsewhere. The get() method is called on controller init and supplies the query id. Then you don't need to use GLOBALS, too.